src/nvim/tui/input.c

```diff
@@ -206,19 +271,46 @@ static void tinput_enqueue(TermInput *input, char *buf, size_t size)
   rbuffer_write(input->key_buffer, buf, size);
 }
 
+static void handle_kitty_key_protocol(TermInput *input, TermKeyKey *key)
+{
+  const char *name = map_get(KittyKey, cstr_t)(&kitty_key_map, (KittyKey)key->code.codepoint);
+  if (name) {
+    char buf[64];
+    size_t len = 0;
+    buf[len++] = '<';
+    if (key->modifiers & TERMKEY_KEYMOD_SHIFT) {
+      len += (size_t)snprintf(buf + len, sizeof(buf) - len, "S-");
+    }
+    if (key->modifiers & TERMKEY_KEYMOD_ALT) {
+      len += (size_t)snprintf(buf + len, sizeof(buf) - len, "A-");
+    }
+    if (key->modifiers & TERMKEY_KEYMOD_CTRL) {
+      len += (size_t)snprintf(buf + len, sizeof(buf) - len, "C-");
+    }
+    len += (size_t)snprintf(buf + len, sizeof(buf) - len, "%s>", name);
+    tinput_enqueue(input, buf, len);
+  }
+}
+
 static void forward_simple_utf8(TermInput *input, TermKeyKey *key)
 {
   size_t len = 0;
   char buf[64];
   char *ptr = key->utf8;
 
-  while (*ptr) {
-    if (*ptr == '<') {
-      len += (size_t)snprintf(buf + len, sizeof(buf) - len, "<lt>");
-    } else {
-      buf[len++] = *ptr;
+  if (key->code.codepoint >= 0xE000 && key->code.codepoint <= 0xF8FF
+      && map_has(KittyKey, cstr_t)(&kitty_key_map, (KittyKey)key->code.codepoint)) {
+    handle_kitty_key_protocol(input, key);
```

Comment:
So this works automatically based on the codepoint? Neither users nor Nvim need to set a terminal option? Would this be potentially usable by any terminal emulator that allows configuring arbitrary "key => sequenc" mappings?

Do we need a doc near `:help tui-input`, similar to `:help tui-csiu` ?